rclpy/rclpy/qos.py

```diff
@@ -343,6 +352,7 @@ def __init__(self, qos_profile: QoSProfile, profile_name: str):
 
 
 class QoSPresetProfiles(Enum):
+    DEFAULT = _qos_profile_default
```

Comment:
Nitpick: Unless there is a strong reason for providing this I think it just causes confusion on what the difference is between `DEFAULT` and `SYSTEM_DEFAULT`. Not to mention that the settings should be the same for them, with the only difference as far as I can tell being that `SYSTEM_DEFAULT` has a `name` attribute set because it's using the `DeprecatedQoSProfile`.

Reply:
That's not quite correct. `DEFAULT` is using `rmw_qos_profile_default`https://github.com/ros2/rmw/blob/master/rmw/include/rmw/qos_profiles.h#L51 - the "chosen default settings for ros2 publishers and subscriptions", whereas `SYSTEM_DEFAULT` is a different qos profile `rmw_qos_profile_system_default` https://github.com/ros2/rmw/blob/master/rmw/include/rmw/qos_profiles.h#L51 - which sets all DDS Policy settings to SYSTEM_DEFAULT, basically using the rmw implementation's default settings.

These two profiles are slightly different - notably (and why this "bug" existed) `rmw_qos_profle_default` chooses `reliability=RELIABLE`, whereas SYSTEM_DEFAULT I think evaluates to `BEST_EFFORT` in Fast-RTPS

Reply:
I do agree, however, that it's confusing and not documented. I think it is worth a pass to document the rmw/qos_profiles.h and places that mirror it e.g. here. However, I think that's a separate change from this one.

Reply:
You're right. I misread the diff. Also agree that updating documentation is separate from this.

Reply:
Using the "default" profile is deprecated, and will be removed in the future. I'd rather not add it as a preset profile, since we discourage it's use.
I understand that the default profile is still in use at the rmw/rcl layers, but I would expect this to be removed in the future (I may be wrong about this). At the very least, we've decided not to allow using it at the client library level.

Reply:
@jacobperron To try and clarify what you're saying - do you mean we're deprecating _exposing_ `rmw_qos_profile_default` to users? Because it seems that we do intend it to be the implicit default, based on the fairly new QoS class in rclcpp - https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/qos.hpp#L62

Reply:
Yes, we are deprecating exposing it to be used directly. Regarding the linked rclcpp code, some settings of the rmw profile (ie. history and depth) are overridden by the `QoSInitialization` object passed as the first argument. The QoSInitialization settings need to be explicitly set by the user.

---

So, while I think initializing `QoSProfile` internally with the default profile is fine, we should not enable users to use the default profile directly, without them being explicit about the history and/or depth setting.

Reply:
Ok, got it. Removing it from `QoSPresetProfiles`. The history/depth requirement is unchanged by this PR, i.e. it still generates warnings when neither is specified.